File: nomad-samples/admin/py/basic-live-channel-1.1/live_channel/get_live_channels.py
from constants.project_constants import *
from exceptions.api_exception_handler import *


import requests, json

def get_live_channels(AUTH_TOKEN):
    API_URL = f"{ADMIN_URL}/liveChannel"

    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        # Send the request
        RESPONSE = requests.get(API_URL, headers= HEADERS)
        
        
        
        # If not found return None
        if (not RESPONSE.ok):
            raise Exception()
        
        

        return RESPONSE.json()


    except:
        api_exception_handler(RESPONSE, f"Get Live Channels failed")

File: nomad-samples/admin/js/basic-live-channel/live-channel/get-live-channels.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";


export default async function getLiveChannels(AUTH_TOKEN) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/liveChannel`, {
        method: "GET",
        headers: HEADERS
    });

    // Check for valid response
    if (RESPONSE) {
        // Check for success
        if (RESPONSE.ok) {
            return await RESPONSE.json();
        }

        // If not found return null
        if (RESPONSE.status === 404) {
            console.error(`Live Channel was not found`);
            return null;
        }
    }

    await apiExceptionHandler(RESPONSE, `Get Live Channel failed`);
}
